Wrappers/Python/cil/processors/FluxNormaliser.py

```diff
@@ -0,0 +1,408 @@
+        
+        data = self.get_input()
+        if angle_index is not None and 'angle' in data.dimension_labels:
+            data_slice = data.get_slice(angle=angle_index)
+        else:
+            data_slice = data
+        
+        if 'channel' in data.dimension_labels:
+            data_slice = data_slice.get_slice(channel=channel_index)
+
+        if len(data_slice.shape) != 2:
+            raise ValueError("Data shape not compatible with preview_configuration(), data must have at least two of 'horizontal', 'vertical' and 'angle'")
+
+        extent = [0, data_slice.shape[1], 0, data_slice.shape[0]]
+        if 'angle' in data_slice.dimension_labels:
+            min_angle = data_slice.geometry.angles[0]
+            max_angle = data_slice.geometry.angles[-1]
+            for i, d in enumerate(data_slice.dimension_labels):
+                if d !='angle':
+                    extent[i*2]=min_angle
+                    extent[i*2+1]=max_angle
+
+        ax1 = plt.subplot(ax)
+        if log:
+            im = ax1.imshow(numpy.log(data_slice.array), cmap='gray',aspect='equal', origin='lower', extent=extent)
+            plt.gcf().colorbar(im, ax=ax1)
+        else:
+            im = ax1.imshow(data_slice.array, cmap='gray',aspect='equal', origin='lower', extent=extent)
+            plt.gcf().colorbar(im, ax=ax1)
+
+        h = data_slice.dimension_labels[1]
+        v = data_slice.dimension_labels[0]
+
+        if h == 'angle':
+            h_min = min_angle
+            h_max = max_angle
+        else:
+            h_min = self.roi[h][0]
+            h_max = self.roi[h][1]
+
+        if v == 'angle':
+            v_min = min_angle
+            v_max = max_angle
+        else:
+            v_min = self.roi[v][0]
+            v_max = self.roi[v][1]
+
+        ax1.plot([h_min, h_max],[v_min, v_min],'--r')
+        ax1.plot([h_min, h_max],[v_max, v_max],'--r')
+
+        ax1.plot([h_min, h_min],[v_min, v_max],'--r')
+        ax1.plot([h_max, h_max],[v_min, v_max],'--r')
+        
+        title = 'ROI'
+        if angle_index is not None:
+            title += ' angle = ' + str(data.geometry.angles[angle_index])
+        if channel_index is not None:
+            title += ' channel = ' + str(channel_index)
+        ax1.set_title(title)
+
+        ax1.set_xlabel(h)
+        ax1.set_ylabel(v)
+
+    def process(self, out=None):
+        self._calculate_flux()
+        self._calculate_target()
+        data = self.get_input()
+
+        if out is None:
+            out = data.copy()
+
+        proj_size = self.v_size*self.h_size
+        num_proj = int(data.array.size / proj_size)
+        
+        f = self.flux
+        for i in range(num_proj):
+            arr_proj = data.array.flat[i*proj_size:(i+1)*proj_size]
+            if len(self.flux.flat) > 1:
+                f = self.flux.flat[i] 
+            arr_proj *= self.target_value/f
+            out.array.flat[i*proj_size:(i+1)*proj_size] = arr_proj
```

Comment:
This would be trivial to add a numba implementation, and here it would speed things up a lot. So I think we should look at that before we merge.